File: python/hissingmicrophone.py
#!/usr/bin/env python
""" An Attempt to solve the Hissing Microphone problem on Kattis """

# This one is really straightforward on python, via any number of mechanisms
import sys
import re

hiss_re = re.compile(r"ss")

if hiss_re.search(sys.stdin.read()):
    print("hiss")
else:
    print("no hiss")

File: src/bin/hissingmicrophone.rs
// Hissing Microphone problem on Kattis
use std::io::{self, BufRead};

fn main() {
    let stdin = io::stdin();

    // Get the Input
    let data = stdin.lock().lines().next().unwrap().unwrap();

    // I doubt it's necessary to do the string conversion, but it's easy.
    if String::from(data).contains("ss") {
        println!("hiss")
    } else {
        println!("no hiss")
    }
}